### DISCUSSION TITLE
Caching contact page

### DISCUSSION THREAD

**Discussion by stephenmeehanuk** [statamic/cms#5940] [General]
Hi, 

I'm using half measure caching, and I've set

```
 'exclude' => [
        '/contact',
    ],
```
in `config/statamic/static_caching.php`

As expected, that excludes the whole page from the static cache, the form still works - but the page is noticeably slower than the other cached pages.

I've looked at the [Cache Tag](https://statamic.dev/tags/cache) docs, but that seem to be more geared to picking parts of the page that should be cached. I want to do the opposite. Cache everything on the page, except the form.

Is there a way to do this?

**Comment by mscruse** [comment]
I am not sure of the answer to your specific question, but have you looked at the Dynamic Token add-on that allows you to use caching with forms? 

https://statamic.com/addons/webographen/dynamic-token

**Reply by stephenmeehanuk** [reply]
This looks good, thanks for sharing.

**Comment by edalzell** [comment]
You want the upcoming `no_cache` tag, likely. But you can also surround the non-form stuff with the `cache` tag if you want.

It shouldn't be thaaaat slow though, what's causing the slow down?

**Reply by stephenmeehanuk** [reply]
Ah, `no_cache` tag sounds like just the ticket! 

It's not rage click away slow, there's just a bit of a lag. Only noticeable really because Statamic does such a great job of caching the other pages - they load super fast.

I'll keep an eye out for any updates that might include the new `no_cache` tag.
